src/HTSPVFS.cpp

```diff
@@ -345,30 +287,26 @@ bool CHTSPVFS::SendFileRead()
   }
 
   if (m == NULL)
-    return false;
+  {
+    tvherror("vfs fileRead failed");
+    return -1;
+  }
 
-  /* Process */
-  if (htsmsg_get_bin(m, "data", &buf, &len))
+  /* Get Data */
+  if (htsmsg_get_bin(m, "data", buffer, &read))
   {
-    htsmsg_destroy(m);
     tvherror("malformed fileRead response: 'data' missing");
-    return false;
-  }
+    read = -1;
 
   /* Store */
-  if (m_buffer.write((unsigned char*)buf, len) != (ssize_t)len)
+  }
+  else
   {
-    htsmsg_destroy(m);
-    tvherror("vfs partial buffer write");
-    return false;
+    memcpy(buf, *buffer, read);
   }
 
-  /* Gradually increase read length */
-  CLockObject lock(m_mutex);
-
-  if (m_currentReadLength * 2 < MAX_READ_LENGTH)
-    m_currentReadLength *= 2;
-
+  /* Cleanup */
   htsmsg_destroy(m);
-  return true;
+
+  return (int)read;
```

Comment:
To make @adamsutton happy maybe we should return `size_t` all the way up the chain and perform the cast to `int` in `client.cpp` (since that's where the `int` requirement comes from)?